### PR TITLE
KLAR-3040 CheckboxTree fix

### PR DESCRIPTION
**Pull request by ignasveleckisnc**
# What

- What was done in this Pull Request
Fixed pre-selected items (children were not auto-selecting)
- How was it before
- Screenshots
- Any other information that may be useful for the dev to review Pull Request

## Testing

- [x] Is this change covered by the unit tests?

- [x] Is this change covered by the integration tests?

- [x] Is this change covered by the automated acceptance tests? (if applicable)

## Compatibility

- [x] Does this change maintain backward compatibility?

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -84,6 +84,7 @@
     "styled-system": "^5.1.5"
   },
   "devDependencies": {
+    "@types/lodash.clonedeep": "^4.5.6",
```
Comment (nowyDEV) [inline package.json:87]: we already have it, some leftover?

Comment (ignasveleckisnc) [reply package.json]: yes, removed it.

File: src/components/checkboxTree/components/RenderComposition.tsx [hunk 2]
```diff
@@ -46,6 +47,14 @@ export function RenderComposition({
   };
 
   React.useEffect(() => {
+    if (
+      parentUid &&
+      selectedList.includes(parentUid) &&
+      !selectedList.includes(uid)
+    ) {
+      setSelectedList([...selectedList, uid]);
```
Comment (nowyDEV) [inline src/components/checkboxTree/components/RenderComposition.tsx:55]: I guess we don't need to evaluate another `if`, right?
```suggestion
      setSelectedList([...selectedList, uid]);
      return
```

File: src/components/checkboxTree/components/RenderComposition.tsx [hunk 3]
```diff
@@ -46,6 +47,14 @@ export function RenderComposition({
   };
 
   React.useEffect(() => {
```
Comment (nowyDEV) [inline src/components/checkboxTree/components/RenderComposition.tsx:49]: Question:
What's the reason to not specify useEffect dependencies? It does not cause any problems?

Comment (ignasveleckisnc) [reply src/components/checkboxTree/components/RenderComposition.tsx:49]: I tested it now, it doesn't cause any problems, but providing at least `selectedList` (the only dynamic part) as a dependency, seems to cut a few not needed re-renders. Adding all missed deps.

**Review by matassp — APPROVED** [review]

**Review by raczyk — APPROVED** [review]

**Pull request merged**